src/components/UserAutomations.tsx

```diff
@@ -1,6 +1,21 @@
+import { api } from "~/utils/api"
+
 const UserAutomations = () => {
+  const userAutomations = api.automation.getUserAutomations.useQuery()
+  const deleteAutomation = api.automation.deleteAutomation.useMutation()
+
   return (
-    <div className="mx-auto mt-5 w-11/12 h-[45%] border-4 border-blue-600">UserAutomations</div>
+    <div className="mx-auto mt-5 w-11/12 h-[45%] border-4 border-blue-600">{userAutomations.data?.length ? userAutomations.data?.map(((automation, i) => (
```

Comment:
is there an idiomatic way to do this with if/else or are you forced to use ternaries inside the brackets there? 

in most contexts nested ternaries (and ternaries that are many lines long) are a code smell because they're hard to read

Reply:
```tsx
<div className="mx-auto mt-5 w-11/12 h-[45%] border-4 border-blue-600">
      {(() => {
        if (userAutomations.data) {
          if (userAutomations.data.length > 0) {
            return userAutomations.data.map((automation, i) => (
              <div className="flex justify-evenly" key={i}>
                <div>{automation.name}</div>
                <div>{automation.desc}</div>
                <div>{automation.status}</div>
                <button
                  onClick={() => {
                    {/* eslint-disable-next-line @typescript-eslint/no-floating-promises */}
                    deleteAutomation.mutate({ hookID: automation.webhookID });
                  }}
                >
                  delete
                </button>
              </div>
            ));
          } else {
            return <div>Waiting for your first creation!</div>;
          }
        } else {
          return <div>loading...</div>;
        }
      })()}
    </div>
```

Reply:
I think this would lead to less code smell.